### ISSUE TITLE
Error parsing AVTransport events when playing Deezer Tracks

### ISSUE THREAD

**Issue by pfischi** [SoCo/SoCo#388]
An error in event handling rountine occurs, when a Deezer track is played. This happens because of an unknown item class "object.container.album.musicAlbum.#DEFAULT" in the event xml string.

As a workaround, i've added a new class to "data_structures.py" to solve the problem.
```
  class DidlMusicAlbumDefault(DidlAlbum):

    item_class = 'object.container.album.musicAlbum.#DEFAULT'
    tag = ''
    _translation = DidlAudioItem._translation.copy()
    _translation.update(
        {
            'artist': ('upnp', 'artist'),
            'genre': ('upnp', 'genre'),
            'producer': ('upnp', 'producer'),
            'toc': ('upnp', 'toc'),
            'album_art_uri': ('upnp', 'albumArtURI'),
        }
    )


```

With best regards,

Stefan

**Comment by KennethNielsen** [comment]
Hi @pfischi and thanks for reporting this.

I'm fairly certain that this class is not part of the DIDL Lite specification and I think that we have seen it before i.e. a particular music service implementing some class that isn't in the spec.

It could probably get messy real fast if we add these music service specific classes every time, so I'm wondering if we couldn't come up with some general solution. It would probably involve an option to lessen the restriction in the from_didl_string function according to one of these options:

 1) In the event of an unknown class, simple say that everything is an object and return that while issuing a warning
 2) Assume that most likely the extra class is a simple extension upon the class one level up in the hierarchic and therefore simple remove the classes one level at a time until we get something we know, return that and issue a warning
 3) Or, to make it real obvious that we got an unknown class, make a UnkonwnObject class and use that (this time without a warning)

I tend to lean towards 2. It is a little bit of guesswork, but I think it matches both cases we have seen and it would probably be what would return the user with the most useful object.

What do you think? @lawrenceakka ?

**Comment by KennethNielsen** [comment]
Hallo @pfischi 

I finally git around to try and fix this in a more general fasion by the method mentioned by @tagdara in #399.

I have tried to implement the fix in this branch (https://github.com/KennethNielsen/SoCo/tree/pound_subclass_fix) requested for pull in #425.

Can you test it?

Regards Kenneth

**Comment by KennethNielsen** [comment]
This should hopefully be fixed with PR #425 which is now merged. Thanks for reporting it.
